src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -909,12 +909,25 @@ private NuGetProject CreateNuGetProject(Project envDTEProject, INuGetProjectCont
 
             // read MSBuild property RestoreProjectStyle which can be set to any NuGet project sytle
             // and pass it on to NugetFactory which can pass it to each NuGet project provider to consume.
-            var restoreProjectStyle = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), "RestoreProjectStyle");
+            var restoreProjectStyle = VsHierarchyUtility.GetMSBuildProperty(VsHierarchyUtility.ToVsHierarchy(envDTEProject), 
+                ProjectSystemProviderContext.RESTORE_PROJECT_STYLE);
```

Comment:
first get IVsHierarchy instance and then pass it to get build properties instead of creating a new IvsHierarchy everytime